Source/WebCore/page/LocalFrame.cpp

```diff
@@ -1406,19 +1407,41 @@ void LocalFrame::setScrollingMode(ScrollbarMode scrollingMode)
 
 #if ENABLE(CONTENT_EXTENSIONS)
 
-static String generateResourceMonitorErrorHTML()
+static String generateResourceMonitorErrorHTML(bool mayUseLightApparance, bool mayUseDarkApparance)
 {
 #if PLATFORM(COCOA) && HAVE(CUSTOM_IFRAME_UNLOADING_HTML)
+#if HAVE(CUSTOM_IFRAME_UNLOADING_HTML_COLOR_SCHEME_READY)
+    return generateResourceMonitorErrorHTMLForCocoa(mayUseLightApparance, mayUseDarkApparance);
+#else
     return generateResourceMonitorErrorHTMLForCocoa();
+#endif
 #else
-    return WEB_UI_STRING("This frame is hidden for using too many system resources.", "Description HTML for frame unloaded by ResourceMonitor");
+    const auto lightAndDarkScheme = ":root { color-scheme: light dark } "_s;
+    const auto onlyDarkScheme = ":root { color-scheme: only dark } "_s;
+    const auto darkStyle = "div { color: white } "_s;
+    const auto empty = ""_s;
+
+    bool mayUseBothApparance = mayUseDarkApparance && mayUseLightApparance;
+
+    const auto& colorScheme = !mayUseDarkApparance ? empty : mayUseLightApparance ? lightAndDarkScheme : onlyDarkScheme;
+    const auto& darkStyleOpen = mayUseBothApparance ? "@media (prefers-color-scheme: dark) { "_s : empty;
+    const auto& darkStyleClose = mayUseBothApparance ? "} "_s : empty;
+
+    return makeString(
+        "<style>"_s,
+        colorScheme, "div { color: black; background-color: gray } "_s, darkStyleOpen, (mayUseDarkApparance ? darkStyle : ""_s), darkStyleClose,
+        "</style><div>"_s,
+        WEB_UI_STRING("This frame is hidden for using too many system resources.", "Description HTML for frame unloaded by ResourceMonitor"),
+        "</div>"_s
+    );
 #endif
 }
 
 void LocalFrame::showResourceMonitoringError()
 {
     RefPtr iframeElement = dynamicDowncast<HTMLIFrameElement>(ownerElement());
-    if (!iframeElement)
+    RefPtr protectedDocument = this->document();
```

Comment:
I think `this->` is unnecessary.

Reply:
Actually no, we want the variable to be named `document`, not `protectedDocument`. We don't have `protected` variables anymore outside lambda capture.